audbcards/core/dataset.py

```diff
@@ -19,13 +19,35 @@
 
 
 class _Dataset:
+    _table_related_cached_properties = ["segment_durations", "segments"]
+    """Cached properties relying on table data.
+
+    Most of the cached properties
+    rely on the dependency table,
+    the header of a dataset,
+    and misc tables used as scheme labels.
+    Some might also need to load filewise or segmented tables,
+    to gather more information.
+    As this process can be slow,
+    or crash,
+    when tables do not fit into memory,
+    we provide the ``load_tables`` argument in ``audbcards.Dataset``
+    to avoid loading of the tables.
```

Comment:
I think it would make sense to explain the mechanics/behavior of this argument on a high, rather conceptual level. Something like this:

### Effect of `load_tables`

#### initial dataset object access - initial caching:

Persistence of table related cached properties on the initial call dependds on the `load_tables` argument as the method `_cached_properties` processes the state of `load_tables`. So a cached dataset might contain table related cached properties or it might not.

#### Second and further acceess to cached object:

if read from cache (second and further access to dataset object), load\_tables=True will do this: if any of the table related cached properties does not exist it is created and persisted to cache by setting cache\_again to True, the newly cached object does now have the table related cached properties. So it and option **to complete an incomplete** dataset objects in the cache. if `load_tables=False`, it will not perform this completion. In other words, the dataset can only be completed once.


Possibly an mention of the discussions in #30 would also help?


The second thing I can think of might be the mention of a usecase: so probably when creating a dataset overview page one might want to use `load_tables`. When you simply want to inspect a dataset quickly on a notebook you might not use it.  So would a kind of "for whom is this useful" depiction make understanding o g the argument easier?

Reply:
> The second thing I can think of might be the mention of a usecase: so probably when creating a dataset overview page one might want to use `load_tables.` When you simply want to inspect a dataset quickly on a notebook you might not use it. So would a kind of "for whom is this useful" depiction make understanding o g the argument easier?

This I have included to some extend in the discussion of the `load_tables` argument description in the docstring of `audbcards.Dataset`:

![image](https://github.com/user-attachments/assets/28730570-e4c5-4f9b-8147-d066c90afd0f)

Hence, I would trying to avoid repeating the text in the comment as well.

Otherwise, I tried to improve the docstring and it now reads:

```python
    """Cached properties relying on table data.

    Most of the cached properties
    rely on the dependency table,
    the header of a dataset,
    and misc tables used as scheme labels.
    Some might also need to load filewise or segmented tables,
    to gather more information.

    Persistence of table related cached properties
    depends on the ``load_tables`` argument
    of :class:`audbcards.Dataset`.
    When ``load_tables`` is ``True``
    :meth:`audbcards.Dataset._cached_properties`
    is asked to cache table related cached properties as well.
    It infers the table related properties
    from ``_table_related_cached_properties``.
    Which means,
    ``_table_related_cached_properties`` has to list all cached properties,
    that will load filewise or segmented tables.

    If a dataset exists in cache,
    but does not store table related cached properties,
    a call to :class:`audbcards.Dataset`
    with ``load_tables`` is ``True``,
    will update the cache.

    """
```

Reply:
Another update:

```python
    """Cached properties relying on table data.

    Most of the cached properties
    rely on the dependency table,
    the header of a dataset,
    and misc tables used as scheme labels.
    Some might also need to load filewise or segmented tables,
    to gather more information.

    Persistence of table related cached properties
    depends on the ``load_tables`` argument
    of :class:`audbcards.Dataset`.
    If ``load_tables`` is ``True``,
    :meth:`audbcards.Dataset._cached_properties`
    is asked to cache table related cached properties as well.
    If ``load_tables`` is ``False``,
    :meth:`audbcards.Dataset._cached_properties`
    is asked to exclude all cached properties,
    listed in ``_table_related_cached_properties``.
    Which means,
    ``_table_related_cached_properties`` has to list all cached properties,
    that will load filewise or segmented tables.

    If a dataset exists in cache,
    but does not store table related cached properties,
    a call to :class:`audbcards.Dataset`
    with ``load_tables`` is ``True``,
    will update the cache.

    """
```

Reply:
>> Which means,
>>  ``_table_related_cached_properties`` has to list all cached properties,

I think a sentence cannot start with a relative pronoun.
I would prefer `This means ...`

Anyhow I  find that this reads much better.  I details well how cache-related objects come into existence.